packages/flutter/test/material/carousel_test.dart

```diff
@@ -1230,6 +1230,47 @@ void main() {
     expect(tester.takeException(), isNull);
   });
 
+  testWidgets('Infinite Scrolling in CarouselView works correctly', (WidgetTester tester) async {
```

Comment:
Can you add tests that verify that the scroll metrics are accurate? What is the ScrollPosition.maxScrollExtent? Can you verify the scroll position as it goes around to the start again?